Sergey cheremshinsky

sergTch opened the pull request:
<h1>Description</h1>

Start - Preprocessing data "spliting columns. Example:
 Embark that has values S, Q, C splited into 
 Embark_S, Embark_Q, Embark_C with values 0 or 1

Train - made function that train 5 models from sklearn and allow them to vote 
 on validation test showes ~82.5% accuracy
 also made neural network using tensorflow with ~86% accuracy

However both tries showes only 76-80% accuracy on kaggle

<h2>How has this been tested?</h2>

- [ ] Executed code few times from start to end without crashing
- [ ] Only ran once
- [ ] Code have dedicated unit tests

sergTch committed:
Add files via upload

sergTch committed:
Add files via upload

sergTch committed:
Delete npFunctions.py

sergTch committed:
Add files via upload

sergTch committed:
Add files via upload

sergTch committed:
Add files via upload

sergTch committed:
Add files via upload

sergTch committed:
Add files via upload

sergTch committed:
Add files via upload

meanalexrin commented on main.py:
```diff
@@ -0,0 +1,156 @@
+    x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.2)
+    x_train = x
+    y_train = y
+    # exit()
```
Usually it's better to remove unused code from the project. It helps readability

meanalexrin commented on main.py:
```diff
@@ -0,0 +1,156 @@
+    print('Exported!')
+
+
+generateAns()
```
You may consider using `if __name__ == "__main__"` in the python code to indicate entry points to the program 

https://stackoverflow.com/a/419185

meanalexrin reviewed (changes requested):
LGTM - Looks Good To Me
Except:
Please move you work to the `/[username]_code` location, as described in readme
https://github.com/rnd4u-org/2021-knu-cairl#development

I'll approve after that
Also that's a good approach that you are using separate .py files rather than a single jupyter notebook

sergTch committed:
Rename main.py to sergey_cheremshinsky/task1(titanic)/main.py

sergTch committed:
Rename ann_model.py to sergey_cheremshinsky/task1(titanic)/ann_model.py

sergTch committed:
Rename sklearn_models.py to sergey_cheremshinsky/task1(titanic)/sklearn_models.py

sergTch committed:
Rename test.csv to sergey_cheremshinsky/task1(titanic)/test.csv

sergTch committed:
Rename titanic.csv to sergey_cheremshinsky/task1(titanic)/titanic.csv

sergTch committed:
Rename train.csv to sergey_cheremshinsky/task1(titanic)/train.csv

sergTch committed:
Add files via upload

sergTch committed:
Rename sergey_cheremshinsky/get_faces.py to sergey_cheremshinsky/task3(mask)/get_faces.py

sergTch committed:
Rename sergey_cheremshinsky/load_data.py to sergey_cheremshinsky/task3(mask)/load_data.py

sergTch committed:
Rename sergey_cheremshinsky/main.py to sergey_cheremshinsky/task3(mask)/main.py

sergTch committed:
Rename sergey_cheremshinsky/log.txt to sergey_cheremshinsky/task3(mask)/log.txt

sergTch committed:
Rename sergey_cheremshinsky/model.py to sergey_cheremshinsky/task3(mask)/model.py

sergTch committed:
Add files via upload

sergTch committed:
Add files via upload

sergTch commented:
I fixed placements of files. Added solution for mask classification.
Didn't add to titanic solution if _name_ == "_main_": but used it in masks
In new project also deleted all code comments however regretted few times. Need to get used to it:)

sergTch commented:
On masks dataset used inception v3 with 1024 512 and 128 dense layers. Trained in 3 steps
freezing 0 : 210 layers, then 0 : 105 and 210 : 316 and last freezed 105 : 316
that helped to train all parts of cnn and achieve 100% on training data and 99.4% on test
At start splited data on test and train where test is 20% of dataset and train is other 80%

meanalexrin commented on sergey_cheremshinsky/task3(mask)/load_data.py:
```diff
@@ -0,0 +1,57 @@
+        y_test = np.load('y_test.npy', allow_pickle=True)
+
+        return (x_train, x_test, y_train, y_test)
+    except Exception:
```
Try to use less general errors or if statements

meanalexrin commented on sergey_cheremshinsky/task3(mask)/main.py:
```diff
@@ -0,0 +1,47 @@
+                epochs=10,
+                validation_data=(x_test, y_test),
+                validation_steps=1511 // batch,
+                steps_per_epoch=6042 // batch,
```
Please add constants and description for magical numbers like 1511 and 6042

meanalexrin commented on sergey_cheremshinsky/task3(mask)/load_data.py:
```diff
@@ -0,0 +1,57 @@
+    x_test = [images[i] for i in indexes[n:]]
+
+    y_train = [labels[i] for i in indexes[:n]]
+    y_test = [labels[i] for i in indexes[n:]]
```
Please use existing train test split function to assure correct execution

https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html

meanalexrin reviewed (changes requested):
Please check other comments

Regarding task 3:
You can try rerun model few times without saving `.npy` indexes. So you'll test model on few different train/test splits

sergTch commented:
I used saving of model and saving of splited data mostly because that's hard for my laptop to finish training in some reasonable amount of time. So I splited training for three nights saving model after each and using same split of data

sergTch committed:
Add files via upload

sergTch committed:
Add files via upload

sergTch committed:
Add files via upload

sergTch committed:
Create main.py

sergTch committed:
Rename sergey_cheremshinsky/task4(Faces)/main.py to sergey_cheremshinsky/task4(faces)/main.py

sergTch committed:
Add files via upload

sergTch committed:
Add files via upload

sergTch commented:
I fixed mistakes that you mentioned about task 3 and also pushed solution for task 4

sergTch committed:
Add files via upload
